src/main/java/uk/gov/hmcts/authorisation/config/ServiceTokenGeneratorConfiguration.java

```diff
@@ -0,0 +1,35 @@
+package uk.gov.hmcts.authorisation.config;
+
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.cloud.netflix.feign.EnableFeignClients;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.context.annotation.Lazy;
+import uk.gov.hmcts.authorisation.ServiceAuthorisationApi;
+import uk.gov.hmcts.authorisation.generators.AuthTokenGenerator;
+import uk.gov.hmcts.authorisation.generators.CachedServiceAuthTokenGenerator;
+import uk.gov.hmcts.authorisation.generators.ServiceAuthTokenGenerator;
+
+@Configuration
+@Lazy
+@EnableFeignClients(basePackageClasses = ServiceAuthorisationApi.class)
+public class ServiceTokenGeneratorConfiguration {
+
+    @Bean
+    public AuthTokenGenerator serviceAuthTokenGenerator(
+            @Value("${idam.s2s-auth.totp_secret}") final String secret,
+            @Value("${idam.s2s-auth.microservice}") final String microService,
+            final ServiceAuthorisationApi serviceAuthorisationApi
+    ) {
+        return new ServiceAuthTokenGenerator(secret, microService, serviceAuthorisationApi);
+    }
+
+    @Bean
+    public AuthTokenGenerator cachedServiceAuthTokenGenerator(
+            @Qualifier("serviceAuthTokenGenerator") AuthTokenGenerator serviceAuthTokenGenerator,
+            @Value("${idam.s2s-auth.tokenTimeToLiveInSeconds:14400}"
+            ) int ttl) {
+        return new CachedServiceAuthTokenGenerator(serviceAuthTokenGenerator, ttl);
+    }
+}
```

Comment:
what is this class for?

Reply:
this caches the token

Reply:
I mean `ServiceTokenGeneratorConfiguration` class

Reply:
if yes, could introduce `ConfigurationProperties` interface and so reduce the character clutter within bean definition

Reply:
done